src/git.ts

```diff
@@ -122,159 +122,123 @@ export class Git implements vscode.Disposable {
     }
 
     public async getBranches(repo: Repository): Promise<Branch[]> {
-        const path = repo.rootUri.fsPath;
-
-        if (!path) {
-            return [];
-        }
-
-        let branchNames: string[];
-
-        try {
-            const {stdout, stderr} = await exec(
-                `${this.gitPath} branch`,
-                {
-                    cwd: path
-                }
-            );
-            branchNames = stdout.split(/\n/g).filter(branch => !!branch);
-        } catch (err) {
-            vscode.window.showErrorMessage('Failed to retrieve the branches\n\n' + (err as any).stderr);
-
-            throw err;
-        }
-
-        const branches: Branch[] = branchNames.map((branch) => {
-            const isStarred = branch.indexOf('*') === 0;
-            const branchName = isStarred ? branch.slice(1).trim() : branch.trim();
-
+        const gitBranches = await repo.getBranches({ remote: false })
+            .then<GitBranch[]>((all) => Promise.all(all.map((b) => repo.getBranch(b.name!!))));
+        const head = repo.state.HEAD;
+        const viewBranches: Branch[] = gitBranches.map((branch) => {
```

Comment:
Previously the branches would be sorted alphabetically by git cmd. The Git extension doesn't seem to keep them sorted.
I'm ok if you want to add a setting to change the sorting method (unsorted/alphabetic/date changed/etc), but at a minimum it would be best if the sorting method stays the same for now.

Reply:
in my test I had always got branches in name order. I had added the sorting so let me know if it does not work for you

Reply:
Branches are sorted correctly for me again. 👍